### PR TITLE
Glib: New version 2.83.0

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by giordano** [comment]
> Glib 2.83.0 is broken on aarch64.

Is there any upstream bug about this?  We've hit lots of corner cases with glib and other GTK-related projects in the past, proactively reporting issues turned out to be important.

**Comment by giordano** [comment]
Also, for the benefit of future readers [the error](https://buildkite.com/julialang/yggdrasil/builds/15253#01938dfd-f1e5-4f4a-802b-07ac0ddb12e1/679-10481) is:
```
[19:59:38] ninja: job failed: /opt/bin/aarch64-linux-gnu-libgfortran3-cxx11/aarch64-linux-gnu-gcc -Iglib/libglib-2.0.so.0.8300.0.p -Iglib -I../glib -I. -I.. -I/workspace/destdir/include -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O3 -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wfloat-conversion -Wmisleading-indentation -Wmissing-field-initializers -Wnonnull -Wnull-dereference -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -fPIC -pthread '-DG_LOG_DOMAIN="GLib"' -DGLIB_COMPILATION -MD -MQ glib/libglib-2.0.so.0.8300.0.p/gutf8.c.o -MF glib/libglib-2.0.so.0.8300.0.p/gutf8.c.o.d -o glib/libglib-2.0.so.0.8300.0.p/gutf8.c.o -c ../glib/gutf8.c
[19:59:38] ../glib/gutf8.c:1933:1: error: ifunc is not supported on this target
[19:59:38]  g_utf8_validate (const char   *str,
[19:59:38]  ^~~~~~~~~~~~~~~
[19:59:38] ../glib/gutf8.c:2019:1: error: ifunc is not supported on this target
[19:59:38]  g_str_is_ascii (const gchar *str)
[19:59:38]  ^~~~~~~~~~~~~~
```
Is that a matter of using a newer GCC?

File: G/Glib/build_tarballs.jl [hunk 1]
```diff
@@ -95,14 +95,14 @@ dependencies = [
     # Host gettext needed for "msgfmt"
     HostBuildDependency("Gettext_jll"),
     Dependency("Libiconv_jll"),
-    Dependency("Libffi_jll", v"3.2.2"; compat="~3.2.2"),
+    Dependency("Libffi_jll", v"3.2.2"; compat="~3.2.2"), # would need 3.4.6 for aarch64-*-freebsd
     # Gettext is only needed on macOS, as far as I could see
     Dependency("Gettext_jll", v"0.21.0"; compat="=0.21.0"),
-    Dependency("PCRE2_jll"; compat="10.35"),
+    Dependency("PCRE2_jll"; compat="10.43"),
```
Comment (giordano) [inline G/Glib/build_tarballs.jl:101]: PCRE2 is a Julia dependency, this would make the package uninstallable

**Comment by eschnett** [comment]
Yes, there is an upstream bug (and a commit, I believe) about the `ifunc` problem. I was just too lazy to turn that commit into a patch.

File: G/Glib/build_tarballs.jl (hunk 1 above)
Comment (eschnett) [reply G/Glib/build_tarballs.jl:101]: Yes, I realized that later.

**Pull request closed without merging**